engine/common/grpc/compressor/deflate/deflate.go

```diff
@@ -0,0 +1,74 @@
+// Package deflate implements and registers the DEFLATE compressor
+// during initialization.
+package deflate
+
+import (
+	"compress/flate"
+	"io"
+	"sync"
+
+	"google.golang.org/grpc/encoding"
+)
+
+// Name is the name registered for the DEFLATE compressor.
+const Name = "deflate"
+
+func init() {
+	c := &compressor{}
+	w, _ := flate.NewWriter(nil, flate.DefaultCompression)
+	c.poolCompressor.New = func() interface{} {
+		return &writer{Writer: w, pool: &c.poolCompressor}
+	}
+	encoding.RegisterCompressor(c)
+}
+
+type writer struct {
```

Comment:
can you organize these files to group the structs with their code? preferably with the struct definition first, followed by its methods.